src/components/admin/Settings/PrefixAlerts/EmailSelector.tsx

```diff
@@ -0,0 +1,227 @@
+import {
+    Autocomplete,
+    AutocompleteRenderInputParams,
+    Chip,
+    FormControl,
+    FormHelperText,
+    ListItem,
+    ListItemText,
+    TextField,
+    Typography,
+} from '@mui/material';
+import { EmailDictionary } from 'components/admin/Settings/PrefixAlerts/types';
+import UserAvatar from 'components/shared/UserAvatar';
+import usePrefixAdministrators from 'hooks/usePrefixAdministrators';
+import useUserInformationByPrefix from 'hooks/useUserInformationByPrefix';
+import { Dispatch, SetStateAction, useMemo, useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { Grant_UserExt } from 'types';
+import { hasLength } from 'utils/misc-utils';
+
+interface Props {
+    prefix: string;
+    emailsByPrefix: EmailDictionary;
+    setEmailsByPrefix: Dispatch<SetStateAction<EmailDictionary>>;
+}
+
+const simpleEmailRegEx = new RegExp(/.+@.+/m);
+
+const minCapability = 'admin';
+
+function EmailSelector({ prefix, emailsByPrefix, setEmailsByPrefix }: Props) {
+    const intl = useIntl();
+
+    const [inputValue, setInputValue] = useState('');
+
+    const { data: adminPrefixes } = usePrefixAdministrators(
+        prefix,
+        minCapability
+    );
+
+    const { data: userInfo } = useUserInformationByPrefix(
+        [prefix, ...adminPrefixes],
+        minCapability
+    );
+
+    const emails = useMemo(
+        () =>
+            Object.hasOwn(emailsByPrefix, prefix) ? emailsByPrefix[prefix] : [],
```

Comment:
This could be made a bit simpler with `emailsByPrefix[prefix] ?? []`

Reply:
That should not be possible since the compiler _should_ complain that the conditional is unnecessary since `emailByPrefix[prefix]` cannot be evaluated to `null` or `undefined` at compile time.